Sources/Basics/Triple+Basics.swift

```diff
@@ -116,6 +116,14 @@ extension Triple {
             )
         }
     }
+
+    public func matches(_ triple: Triple) -> Bool {
+        guard self.isMacOSX, let version = self._macOSVersion, let comparedVersion = triple._macOSVersion else {
+            return self.tripleString == triple.tripleString
```

Comment:
This seems wrong to me, is that how equality works in the swift-driver triple that we're using now? If yes, I guess we have a huge regression waiting to happen. In TSC's triple, we had this https://github.com/apple/swift-tools-support-core/blob/main/Sources/TSCUtility/Triple.swift#L234

The reason for this is that is that triples may have different literal spellings even though they're semantically the same e.g. macos vs. macosx

Reply:
`isMacOSX` actually does that canonicalization for us.

Reply:
Sure, but that's only one special case, I don't think we should ever compare the string representation of triples.

Reply:
Rewrote the check so that it no longer does string representation comparison.